### PR TITLE
Add automatic remote uploader downloader for composer profiler

### PR DESCRIPTION
**Pull request by j316chuck**
# What does this PR do?
Add profiling remote downloader to composer

### CODE REVIEW

**Comment by j316chuck** [comment]
@eracah any advice on how to add a test for this?

File: composer/trainer/trainer.py [hunk 1]
```diff
@@ -1035,15 +1035,19 @@ def __init__(
...
+        # Console Logging
+        loggers = list(ensure_tuple(loggers))
+
         # Profiler
         if profiler is not None:
             warnings.warn('The profiler is enabled. Using the profiler adds additional overhead when training.')
             self.state.profiler = profiler
+            for remote_uri in profiler.remote_filenames:
+                remote_ud = maybe_create_remote_uploader_downloader_from_uri(uri=remote_uri, loggers=loggers)
+                if remote_ud:
```
Comment (dakinggg) [inline composer/trainer/trainer.py:1047]:
```suggestion
                if remote_ud is not None:
```

File: composer/profiler/profiler.py [hunk 2]
```diff
@@ -108,6 +110,21 @@ def __init__(
         self.schedule = schedule
         self.state = None
         self._callbacks: List[Callback] = []
+        self.remote_filenames: List[str] = []
+        # First, add each remote file name to self.remote_filenames to create RemoteUplaoderDownloader logger in trainer. [s3://bucket/path/to/file]
```
Comment (dakinggg) [inline composer/profiler/profiler.py:114]:
```suggestion
        # First, add each remote file name to self.remote_filenames to create RemoteUploaderDownloader logger in trainer. [s3://bucket/path/to/file]
```

File: composer/profiler/profiler.py [hunk 3]
```diff
@@ -108,6 +110,21 @@ def __init__(
...
+        # Then modify remote file name to be a local path to pass into torch_profiler and system_profiler. e.g: path/to/file
+        if torch_prof_remote_file_name:
+            self.remote_filenames.append(torch_prof_remote_file_name)
+            _, _, torch_prof_remote_file_name = parse_uri(torch_prof_remote_file_name)
```
Comment (dakinggg) [inline composer/profiler/profiler.py:118]: this seems unused?

Comment (j316chuck) [reply composer/profiler/profiler.py:118]: Used here: https://github.com/mosaicml/composer/pull/2653/files/b29a586e15a4ae60a77fe4b0f075abaf44811a24#diff-f264eab2433404bea45e45f53b2562cf2b3083199c4618873d973ca43adb1334R141

**Review by eracah — APPROVED** [review]
LGTM, but note that:
1. all the torch_folder/filename arguments can be reduced to 1 for uri parsing
2. logger.upload_file in torchprofiler will upload profile traces to all object paths including ones used for checkpointing

**Pull request merged**